lib/statue/sidekiq_statistics.rb

```diff
@@ -0,0 +1,73 @@
+require 'statue'
+
+module Statue
+  # This module simplifies the job statistics tracking using Statue
+  # * Provides a SidekiqMiddleware (to track performance and latency)
+  # * Provides a method for tracking other middleware events (eg. throttling, deadline)
+  #
+  # The current set of metrics are:
+  #
+  # * count job.<queue>.<job_name>.(success|failure):
+  #   depending if the job succeeded or failed
+  # * count job.<queue>.<job_name>.throttled:
+  #   only if the job was throttled by sidekiq-throttler
+  # * count job.<queue>.<job_name>.overdue:
+  #   only if the deadline from run_deadline_middleware was reached
+  # * count job.<queue>.<job_name>.retry:
+  #   only if the job corresponds to a retry for a previously failed job
+  # * duration job.<queue>.<job_name>.latency:
+  #   time difference between now and when the job last entered the queue
+  # * duration job.<queue>.<job_name>:
+  #   job run duration (reported only if the job doesn't fail)
+  module SidekiqStatistics
```

Comment:
Les copa este nombre? O prefieren `JobStatistics`?

Reply:
This middleware hooks itself to a **Sidekiq**'s middleware, not sure if other job runners are compatible.